File: Examples/CSharp/GroupDocs.Text.Examples.CSharp/EmailsExtractor.cs
using GroupDocs.Text;
using GroupDocs.Text.Extractors.Text;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GroupDocs.Text_for_.NET
{
    public class EmailsExtractor
    {
        public static void ExtractEmailAttachments(string fileName)
        {
            //ExStart:ExtractEmailAttachments
            //get file actual path
            String filePath = Utilities.getFilePath(fileName);
            EmailTextExtractor extractor = new EmailTextExtractor(filePath);
            ExtractorFactory factory = new ExtractorFactory();
            for (int i = 0; i < extractor.AttachmentCount; i++)
            {
                Console.WriteLine(extractor.GetContentType(i).Name);
                Stream stream = extractor.GetStream(i);
                TextExtractor attachmentExtractor = factory.CreateTextExtractor(filePath);
                try
                {
                    Console.WriteLine(attachmentExtractor == null ? "Document format is not supported" : attachmentExtractor.ExtractAll());
                }
                finally
                {
                    if (attachmentExtractor != null)
                    {
                        attachmentExtractor.Dispose();
                    }
                }
            }
            //ExEnd:ExtractEmailAttachments
        }
    }
}

File: Examples/VBasic/GroupDocs.Text.Examples.VBasic/EmailsExtractor.vb
Imports System.IO
Imports GroupDocs.Text.Extractors.Text

Public Class EmailsExtractor


    Public Shared Sub ExtractEmailAttachments(fileName As String)
        'ExStart:ExtractEmailAttachments
        'get file actual path
        Dim filePath As [String] = Common.getFilePath(fileName)
        Dim extractor As New EmailTextExtractor(filePath)
        Dim factory As New ExtractorFactory()
        For i As Integer = 0 To extractor.AttachmentCount - 1
            Console.WriteLine(extractor.GetContentType(i).Name)
            Dim stream As Stream = extractor.GetStream(i)
            Dim attachmentExtractor As TextExtractor = factory.CreateTextExtractor(filePath)
            Try
                Console.WriteLine(If(attachmentExtractor Is Nothing, "Document format is not supported", attachmentExtractor.ExtractAll()))
            Finally
                If attachmentExtractor IsNot Nothing Then
                    attachmentExtractor.Dispose()
                End If
            End Try
        Next
        'ExEnd:ExtractEmailAttachments
    End Sub

End Class
